train.py

```diff
@@ -102,9 +102,13 @@
     help='Which metric to use for the distance between embeddings.')
 
 parser.add_argument(
-    '--loss', default='batch_hard', choices=loss.LOSS_CHOICES.keys(),
+    '--loss', default='batch_hard', choices=loss.LOSS_CHOICES,
     help='Enable the super-mega-advanced top-secret sampling stabilizer.')
 
+parser.add_argument(
+    '--loss_ignore_zero', default=False, const=True, nargs='?', type=common.positive_float,
```

Comment:
Ohh misread this to mean it can only be true or false. I am going to start playing with this then.